pkg/metrics/metrics.go

```diff
@@ -0,0 +1,63 @@
+package metrics
+
+import "github.com/prometheus/client_golang/prometheus"
+
+const (
+	separator = "_"
+
+	subsystem = "stonesoup_sprayproxy"
+
+	github                    = "github"
+	githubRequestsName        = subsystem + separator + github + separator + "requests_total"
+	forwarded                 = "forwarded"
+	forwardedRequestsName     = subsystem + separator + forwarded + separator + "requests_total"
+	responseTime              = "response" + separator + "time"
+	forwardedResponseTimeName = subsystem + separator + responseTime + separator + "duration_seconds"
+
+	MetricsPort = 6000
+)
+
+var (
+	githubRequestCounter, forwardedRequestCounter = createCounters()
+	responseTimeHistogram                         = createHistogram()
+)
+
+func init() {
+	prometheus.MustRegister(githubRequestCounter)
+	prometheus.MustRegister(forwardedRequestCounter)
+	prometheus.MustRegister(responseTimeHistogram)
+}
+
+func createCounters() (prometheus.Counter, prometheus.Counter) {
+	return prometheus.NewCounter(prometheus.CounterOpts{
+			Name: githubRequestsName,
+			Help: "Counts incoming requests from github.",
+			//TODO do we want to add a label for request URL, or is that too numerous?
+		}),
+		prometheus.NewCounter(prometheus.CounterOpts{
+			Name: forwardedRequestsName,
+			Help: "Counts forwarded attempts to backend server(s).",
+			//TODO do we want to add a label for backend server?
+		})
+}
+
+func createHistogram() prometheus.Histogram {
+	return prometheus.NewHistogram(prometheus.HistogramOpts{
+		Name: forwardedResponseTimeName,
+		Help: "Forwarded request duration in seconds.",
+		//TODO do we want to allow configuration of buckets?
```

Comment:
Yes - let's start with an SLO of "95% of forwarded requests should complete in under 5 seconds". I'd use an exponential bucket scheme:

```suggestion
                // Create buckets of 0.005, 0.05, 0.5, 5, and +Infinity
		Buckets: prometheus.ExponentialBuckets(0.005, 10, 4)
```

Reply:
part of next push